crates/doc/src/combine/spill.rs

```diff
@@ -369,18 +369,18 @@ impl<F: io::Read + io::Seek> Iterator for SpillDrainer<F> {
             let key = &self.spec.keys[binding as usize];
             let (validator, ref schema) = &mut self.spec.validators[binding as usize];
 
-            // LazyNode root which is updated as reductions occur.
-            let mut root = LazyNode::Node(owned_root.get());
+            // Reduced HeapNode which is updated as reductions occur.
+            let mut root: Option<HeapNode<'_>> = None;
```

Comment:
LazyNode no longer "owns" its HeapNode so that ownership now needs to live elsewhere (here). Instead, we create LazyNode instances on the fly by-reference below.

> was that the whole point was to allow nodes that were very large to spill to disk

Kinda. We don't spill individual nodes, but we do spill batches of them. Here, we're working with either a HeapNode, or a previously spilled ArchivedNode which has been read into aligned memory, and can be walked without further deserialization as an &ArchivedNode.